### ISSUE TITLE
Unification creates cyclic term

### ISSUE THREAD

**Issue by pmoura** [terminusdb/terminus-server#106]
In the following goal, the unification creates cyclic term:

https://github.com/terminusdb/terminus-server/blob/4c2dbdd4c3b46d7a0f1d5c470c7f6132ef99238a/library/validate.pl#L170-L173

Reported by the Logtalk linter as:

```text
*     Suspicious call: repository_descriptor{database_descriptor:A,repository_name:B}=A as unification will succeed creating a cyclic term
*       while compiling object validate
*       in file /Users/pmoura/terminus-server/library/validate.pl between lines 164-187
```

**Comment by GavinMendelGleason** [comment]
Woops! We'll fix that. Cheers.

**Comment by pmoura** [comment]
Fixed in cdb23f07b46aede2d2775128b9ea598b922f8d12.
